ci2/ci/build.py

```diff
@@ -316,60 +352,80 @@ def config(self):  # pylint: disable=no-self-use
         else:
             output_files = None
 
-        sa = None
-        if (self.outputs is not None) or (self.inputs is not None):
-            sa = 'ci2'
+        volumes = []
+        if self.secrets:
+            for secret in self.secrets:
+                name = secret['name']
+                mount_path = secret['mountPath']
+                volumes.append({
+                    'volume': {
+                        'name': name,
+                        'secret': {
+                            'optional': False,
+                            'secretName': name
+                        }
+                    },
+                    'volume_mount': {
+                        'mountPath': mount_path,
+                        'name': name
+                    }
+                })
 
         self.job = await batch.create_job(
             self.image,
             command=['bash', '-c', rendered_script],
             attributes={'name': self.name},
             input_files=input_files,
             output_files=output_files,
-            copy_service_account_name=sa,
-            parent_ids=self.deps_parent_ids())
+            volumes=volumes,
+            parent_ids=self.deps_parent_ids(),
+            always_run=self.always_run)
```

Comment:
why is the RunImageStep always_run? If the image build fails, we shouldn't bother running it right?

Reply:
`always_run` is for cleanup steps, namely, they run whether the parents (or ancestors) run successfully or not.  So we set `always_run` explicitly on for ci to clean up after itself (e.g. delete temporary images, files, namespaces, etc.)

The user also might want to cleanup in the testing pipeline. For example, testing ci, we want to create a temporary github repo that should be cleaned up after the tests.

Skipping if the image fails is reasonable, although I don't think it hurts to run the job and have it fail, and it isn't easy to do this in batch right now.